CONTRIBUTING.md

```diff
@@ -65,19 +65,52 @@ Thus:
 - if the list includes `next` but no release tags: the fix/enhancement will come in the nearest minor release.
 - if the list includes `main` but no release tags: the fix/enhancement will come in the nearest patch release.
 
+## Proposing new changes
+
+If you want to add functionality and features that don't fall under an existing and accepted issue please add a [Kubernetes Enhancement Proposal (KEP)][keps] in a pull request to the `railgun/keps/` directory.
+
+To the best of your ability: take some time before you propose something new to search through some of the existing KEPs to make sure that your proposal isn't a duplicate nor is it colliding with anything others.
+
+Read our [KEP Documentation][kepstart] which provides some guidelines about how to get started and contributing to new or existing KEPs.
+
+[keps]:https://github.com/kubernetes/enhancements/blob/master/keps/README.md
+[kepstart]:https://github.com/Kong/kubernetes-ingress-controller/blob/next/railgun/keps/README.md
+
 ## Development environment
 
 ## Environment
 
 - Golang version matching our [`Dockerfile`](./Dockerfile) installed
-- Access to a k8s cluster, you can use Minikube or GKE
-- make
-- Docker (for building)
+- [Kubebuilder][kubebuilder] `v3.1.x`+
+- [GNU Make][make] `v4.x`+
+- [Docker][docker] (for building) `v20.10.x`+
+- Access to a k8s cluster, you can use [KIND][kind] [Minikube][minikube] or [GKE][gke]
+
+[kubebuilder]:https://kubebuilder.io/
+[make]:https://www.gnu.org/software/make/
+[docker]:https://docs.docker.com/
+[kind]:https://github.com/kubernetes-sigs/kind
+[minikube]:https://github.com/kubernetes-sigs/minikube
+[gke]:https://cloud.google.com/kubernetes-engine/
 
 ## Dependencies
 
 The build uses dependencies are managed by [go modules](https://blog.golang.org/using-go-modules)
 
+## Developing
+
+Development of our [Kubernetes Controllers][ctrl] and [APIs][kapi] is managed through the [Kubebuilder SDK][kubebuilder].
+
+Prior to developing we recommend you run through the [Kubebuilder Quickstart Documentation][kbquick] documentation in order to familiarize yourself with how the command line works, how to add new APIs and controllers, and how to update existing APIs.
```

Comment:
Although we use the Kubebuilder tooling under the hood, adding APIs also requires familiarity with our Make targets (which we want to simplify and better document in #1358) and code generation sources (`railgun/hack/generators/controllers/networking`).

Reply:
```suggestion
Prior to developing we recommend you read through the [Makefile](/Makefile) directives related to generation of API configurations, and run through the [Kubebuilder Quickstart Documentation][kbquick] documentation in order to familiarize yourself with how the command line works, how to add new APIs and controllers, and how to update existing APIs.
```